src/applications/bmqtool/m_bmqtool_interactive.cpp

```diff
@@ -85,6 +87,104 @@ void printMessage(bsl::ostream& out, int index, const bmqa::Message& message)
 #endif
 }
 
+void closeFileStream(bsl::ifstream* fileStream)
+{
+    fileStream->close();
+}
+
+// Load message content from dump file, created by
+// QueueEngineUtil::dumpMessageInTempfile() method. Return `true` on success or
+// `false` otherwise. In case of `false`, `errorDescription` contains details.
+bool loadMessageContentFromFile(
+    bsl::vector<bsl::string>*     payload,
+    bsl::vector<MessageProperty>* messageProperties,
+    bsl::ostream&                 errorDescription,
+    bsl::string&                  file)
+{
+    if (file.empty()) {
+        errorDescription << "Empty 'file' argument";
+        return false;  // RETURN
+    }
+
+    bsl::ifstream fileStream(file.c_str());
+    if (!fileStream.is_open()) {
+        errorDescription << "Failed to open file: " << file;
+        return false;  // RETURN
+    }
+
+    // Set guard to close the filestream
+    bdlb::ScopeExitAny guard(
+        bdlf::BindUtil::bind(closeFileStream, &fileStream));
+
+    // Parse file according to format defined in
+    // QueueEngineUtil::dumpMessageInTempfile()
+    bsl::string line;
+    bsl::getline(fileStream, line);
+    if (line == "Message Properties:") {
+        char tmpBuffer[2];
+        fileStream.read(tmpBuffer, 1);  // skip empty line
+
+        // Properties
+        bsl::getline(fileStream, line);
+        if (line.at(0) != '[') {
+            errorDescription << "Properties '[' marker missed";
+            return false;  // RETURN
+        }
+        mwcu::MemOutStream propertiesStream;
+        propertiesStream << line;
+        if (line.back() != ']') {
+            // Binary properties are multiline, read lines until close marker
+            // ']'
+            propertiesStream << '\n';
+            while (!fileStream.eof()) {
+                bsl::getline(fileStream, line);
+                propertiesStream << line;
+                // Check for close marker
+                if (line.back() == ']')
+                    break;  // BREAK
+                propertiesStream << '\n';
+            }
+            if (fileStream.eof()) {
+                errorDescription << "Properties ']' marker missed";
+                return false;  // RETURN
+            }
+        }
+
+        bsl::string parseError;
+        if (!InputUtil::parseProperties(messageProperties,
+                                        propertiesStream.str(),
+                                        &parseError)) {
+            errorDescription << "Message properties parse error: "
+                             << parseError;
+            return false;
```

Comment:
Here, and in other places below
```suggestion
            return false;  // RETURN
```